### ISSUE TITLE
Compilation freezes at npm install

### ISSUE THREAD

**Issue by Giszmo** [BlueWallet/BlueWallet#6409]
I tried to compile version 6.6.1 four times now and it always freezes at:

```
$ ./test.sh --apk ~/Documents/walletscrutiny/incoming/BlueWallet\ 6.6.1.apk 
+++ dirname ./test.sh
++ cd .
++ pwd
+ SCRIPT_DIR=/home/leo/StudioProjects/giszmo.github.io/scripts
+ TEST_ANDROID_DIR=/home/leo/StudioProjects/giszmo.github.io/scripts/test/android
+ wsContainer=docker.io/walletscrutiny/android:5
+ takeUserActionCommand='echo "CTRL-D to continue";
  bash'
+ shouldCleanup=false
+ [[ 2 -gt 0 ]]
+ case $1 in
+ downloadedApk='/home/leo/Documents/walletscrutiny/incoming/BlueWallet 6.6.1.apk'
+ shift
+ shift
+ [[ 0 -gt 0 ]]
+ [[ /home/leo/Documents/walletscrutiny/incoming/BlueWallet 6.6.1.apk =~ ^/.* ]]
+ '[' '!' -f '/home/leo/Documents/walletscrutiny/incoming/BlueWallet 6.6.1.apk' ']'
++ sha256sum '/home/leo/Documents/walletscrutiny/incoming/BlueWallet 6.6.1.apk'
++ awk '{print $1;}'
+ appHash=65840ba3a127285f3f76f6471dfe004c4ad9a213b17f56eb3a2d41a4eaf58831
+ fromPlayFolder=/tmp/fromPlay65840ba3a127285f3f76f6471dfe004c4ad9a213b17f56eb3a2d41a4eaf58831
+ rm -rf /tmp/fromPlay65840ba3a127285f3f76f6471dfe004c4ad9a213b17f56eb3a2d41a4eaf58831
++ getSigner '/home/leo/Documents/walletscrutiny/incoming/BlueWallet 6.6.1.apk'
+++ dirname '/home/leo/Documents/walletscrutiny/incoming/BlueWallet 6.6.1.apk'
++ DIR=/home/leo/Documents/walletscrutiny/incoming
+++ basename '/home/leo/Documents/walletscrutiny/incoming/BlueWallet 6.6.1.apk'
++ BASE='BlueWallet 6.6.1.apk'
+++ podman run --rm --volume /home/leo/Documents/walletscrutiny/incoming:/mnt:ro --workdir /mnt docker.io/walletscrutiny/android:5 apksigner verify --print-certs 'BlueWallet 6.6.1.apk'
+++ grep 'Signer #1 certificate SHA-256'
+++ awk '{print $6}'
++ s=42250147991337ed230fbd93c0be0e5f6183d02eed9e1d53e5aac94167cf3f2f
++ echo 42250147991337ed230fbd93c0be0e5f6183d02eed9e1d53e5aac94167cf3f2f
+ signer=42250147991337ed230fbd93c0be0e5f6183d02eed9e1d53e5aac94167cf3f2f
+ echo 'Extracting APK content ...'
Extracting APK content ...
+ containerApktool /tmp/fromPlay65840ba3a127285f3f76f6471dfe004c4ad9a213b17f56eb3a2d41a4eaf58831 '/home/leo/Documents/walletscrutiny/incoming/BlueWallet 6.6.1.apk'
+ targetFolder=/tmp/fromPlay65840ba3a127285f3f76f6471dfe004c4ad9a213b17f56eb3a2d41a4eaf58831
+ app='/home/leo/Documents/walletscrutiny/incoming/BlueWallet 6.6.1.apk'
++ dirname /tmp/fromPlay65840ba3a127285f3f76f6471dfe004c4ad9a213b17f56eb3a2d41a4eaf58831
+ targetFolderParent=/tmp
++ basename /tmp/fromPlay65840ba3a127285f3f76f6471dfe004c4ad9a213b17f56eb3a2d41a4eaf58831
+ targetFolderBase=fromPlay65840ba3a127285f3f76f6471dfe004c4ad9a213b17f56eb3a2d41a4eaf58831
++ dirname '/home/leo/Documents/walletscrutiny/incoming/BlueWallet 6.6.1.apk'
+ appFolder=/home/leo/Documents/walletscrutiny/incoming
++ basename '/home/leo/Documents/walletscrutiny/incoming/BlueWallet 6.6.1.apk'
+ appFile='BlueWallet 6.6.1.apk'
+ podman run --rm --volume /tmp:/tfp --volume /home/leo/Documents/walletscrutiny/incoming:/af:ro docker.io/walletscrutiny/android:5 sh -c 'apktool d -o "/tfp/fromPlay65840ba3a127285f3f76f6471dfe004c4ad9a213b17f56eb3a2d41a4eaf58831" "/af/BlueWallet 6.6.1.apk"'
I: Using Apktool 2.4.1 on BlueWallet 6.6.1.apk
I: Loading resource table...
I: Decoding AndroidManifest.xml with resources...
I: Loading resource table from file: /root/.local/share/apktool/framework/1.apk
I: Regular manifest package...
I: Decoding file-resources...
I: Decoding values */* XMLs...
I: Baksmaling classes.dex...
I: Baksmaling classes2.dex...
I: Baksmaling classes3.dex...
I: Copying assets and libs...
I: Copying unknown files...
I: Copying original files...
I: Copying META-INF/services directory
+ return 0
++ cat /tmp/fromPlay65840ba3a127285f3f76f6471dfe004c4ad9a213b17f56eb3a2d41a4eaf58831/AndroidManifest.xml
++ head -n 1
++ sed 's/.*package=\"//g'
++ sed 's/\".*//g'
+ appId=io.bluewallet.bluewallet
++ cat /tmp/fromPlay65840ba3a127285f3f76f6471dfe004c4ad9a213b17f56eb3a2d41a4eaf58831/apktool.yml
++ grep versionName
++ sed 's/.*\: //g'
++ sed 's/'\''//g'
+ versionName=6.6.1
++ cat /tmp/fromPlay65840ba3a127285f3f76f6471dfe004c4ad9a213b17f56eb3a2d41a4eaf58831/apktool.yml
++ grep versionCode
++ sed 's/.*\: //g'
++ sed 's/'\''//g'
+ versionCode=1712788857
+ workDir=/tmp/test_io.bluewallet.bluewallet
+ '[' -z io.bluewallet.bluewallet ']'
+ '[' -z 6.6.1 ']'
+ '[' -z 1712788857 ']'
+ echo

+ echo 'Testing "/home/leo/Documents/walletscrutiny/incoming/BlueWallet 6.6.1.apk" (io.bluewallet.bluewallet version 6.6.1)'
Testing "/home/leo/Documents/walletscrutiny/incoming/BlueWallet 6.6.1.apk" (io.bluewallet.bluewallet version 6.6.1)
+ echo

+ testScript=/home/leo/StudioProjects/giszmo.github.io/scripts/test/android/io.bluewallet.bluewallet.sh
+ '[' '!' -f /home/leo/StudioProjects/giszmo.github.io/scripts/test/android/io.bluewallet.bluewallet.sh ']'
+ source /home/leo/StudioProjects/giszmo.github.io/scripts/test/android/io.bluewallet.bluewallet.sh
++ repo=https://github.com/BlueWallet/BlueWallet
++ tag=v6.6.1
++ builtApk=/tmp/test_io.bluewallet.bluewallet/app-release-unsigned.apk
+ prepare
+ echo 'Testing io.bluewallet.bluewallet from https://github.com/BlueWallet/BlueWallet revision v6.6.1 (revisionOverride: '\'''\'')...'
Testing io.bluewallet.bluewallet from https://github.com/BlueWallet/BlueWallet revision v6.6.1 (revisionOverride: '')...
+ rm -rf /tmp/test_io.bluewallet.bluewallet
+ mkdir -p /tmp/test_io.bluewallet.bluewallet
+ cd /tmp/test_io.bluewallet.bluewallet
+ echo 'Trying to clone …'
Trying to clone …
+ '[' -n '' ']'
+ git clone --quiet --branch v6.6.1 --depth 1 https://github.com/BlueWallet/BlueWallet app
+ cd app
++ git log -n 1 --pretty=oneline
++ sed 's/ .*//g'
+ commit=749351f3019a5624bce7b315166c49911ed1c0b8
+ test
+ podman rmi bluewallet -f
++ id -u
+ podman build --tag bluewallet --cgroup-manager cgroupfs --ulimit nofile=16384:16384 --build-arg UID=1000 --build-arg TAG=v6.6.1 --build-arg VERSION=1712788857 --file /home/leo/StudioProjects/giszmo.github.io/scripts/test/android/io.bluewallet.bluewallet.dockerfile
STEP 1/11: FROM docker.io/node:18-bullseye-slim
STEP 2/11: ARG UID=1000
--> Using cache ccdafc78060910ae95b92e2986e7f1a5bcfe9595c7dada62b62b2eb532835017
--> ccdafc78060
STEP 3/11: ARG TAG
--> Using cache be119e25a7713072a84fa0ffce28bbe4a4b0bbd21773cc6115b1dd4c2b8a5865
--> be119e25a77
STEP 4/11: ARG VERSION
--> Using cache 7c62c6d6305ec4ad091af6b6e6ee7798e2287bdf1310f228a6eb1fbadd45a42f
--> 7c62c6d6305
STEP 5/11: RUN set -ex;     apt-get update;     DEBIAN_FRONTEND=noninteractive apt-get install --yes       -o APT::Install-Suggests=false --no-install-recommends       patch git openjdk-11-jre-headless openjdk-11-jdk;     rm -rf /var/lib/apt/lists/*;     deluser node;     useradd --uid $UID --create-home --shell /bin/bash appuser;     mkdir -p /Users/runner/work/1/;     chown -R appuser:appuser /Users/;
--> Using cache ed067ea7d4ca0704547eeeae7b7ed9d91dde4b9ce39e247d469c36262a2a8c46
--> ed067ea7d4c
STEP 6/11: USER appuser
--> Using cache 67eb79cb336b36e613edea3c26b1bf095ae5a4129ebe6eb30b39e53ca7c363a0
--> 67eb79cb336
STEP 7/11: ENV ANDROID_SDK_ROOT="/home/appuser/sdk"     ANDROID_HOME="/home/appuser/sdk"     NODE_ENV="production"
--> Using cache be4da7dce03ea2a31d830b746923e56325819030f6bcc3364a521c7b81b4bd62
--> be4da7dce03
STEP 8/11: RUN set -ex;     mkdir -p "/home/appuser/sdk/licenses";     printf "\n24333f8a63b6825ea9c5514f83c2829b004d1fee" > "/home/appuser/sdk/licenses/android-sdk-license";     cd /Users/runner/work/1/;     git clone --branch $TAG https://github.com/BlueWallet/BlueWallet /Users/runner/work/1/s/;
--> Using cache fded534bce3c310636f7061a32122ad9ae08cb37edeca60e5e49c89c6f33f6f0
--> fded534bce3
STEP 9/11: WORKDIR /Users/runner/work/1/s/
--> Using cache e7483f57839f59f3a2b872a42d227fbb058024b3698abe42e5a4634034a6a150
--> e7483f57839
STEP 10/11: RUN set -ex;     npm install --production --no-optional --omit=optional --no-audit --no-fund --ignore-scripts;     npm run postinstall;     rm -rf node_modules/realm; npm install realm;     echo '"master"' > current-branch.json;
+ npm install --production --no-optional --omit=optional --no-audit --no-fund --ignore-scripts
npm WARN config production Use `--omit=dev` instead.
npm WARN config optional Use `--omit=optional` to exclude optional dependencies, or
npm WARN config `--include=optional` to include them.
npm WARN config 
npm WARN config     Default value does install optional deps unless otherwise omitted.
npm WARN skipping integrity check for git dependency ssh://git@github.com/BlueWallet/react-native-prompt-android.git 
npm WARN skipping integrity check for git dependency ssh://git@github.com/BlueWallet/react-native-privacy-snapshot.git 
npm WARN skipping integrity check for git dependency ssh://git@github.com/BlueWallet/react-native-passcode-auth.git 
npm WARN skipping integrity check for git dependency ssh://git@github.com/BlueWallet/react-native-obscure.git 
npm WARN skipping integrity check for git dependency ssh://git@github.com/BlueWallet/react-native-idle-timer.git 
npm WARN skipping integrity check for git dependency ssh://git@github.com/BlueWallet/react-native-handoff.git 
npm WARN skipping integrity check for git dependency ssh://git@github.com/BlueWallet/react-native-document-picker.git 
npm WARN skipping integrity check for git dependency ssh://git@github.com/BlueWallet/localized-strings.git 
npm WARN skipping integrity check for git dependency ssh://git@github.com/BlueWallet/bip47.git 
npm WARN deprecated uglify-es@3.3.9: support for ECMAScript is superseded by `uglify-js` as of v3.13.0
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated @babel/plugin-proposal-object-rest-spread@7.20.7: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-object-rest-spread instead.
npm WARN deprecated @babel/plugin-proposal-numeric-separator@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-numeric-separator instead.
npm WARN deprecated @babel/plugin-proposal-async-generator-functions@7.20.7: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-async-generator-functions instead.
npm WARN deprecated @babel/plugin-proposal-nullish-coalescing-operator@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-nullish-coalescing-operator instead.
npm WARN deprecated @babel/plugin-proposal-class-properties@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-class-properties instead.
npm WARN deprecated @babel/plugin-proposal-optional-catch-binding@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-optional-catch-binding instead.
npm WARN deprecated @babel/plugin-proposal-optional-chaining@7.21.0: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-optional-chaining instead.
```

This looks slightly different than #5700. As my test script does a fresh checkout and run inside that folder, I don't think `npm run clean` would makes sense given it's a "clean" folder.

**Comment by Overtorment** [comment]
how exactly you compile it?

PS. npm install doesnt do nothing fancy. Id say npm mirrors had temp outage or smth

**Comment by Giszmo** [comment]
Ok, this took me many hours and I still don't know why it just freezes instead of timing out but here are my findings:

Using podman instead of docker freezes while using docker it does show timeouts and the script finishes with a build error.

Using pnpm instead of npm, build is much faster and does not run into timeouts but it then is missing several dependencies. I suspect pnpm is somewhat stricter in the dependencies it does install but maybe there is some flag to install what npm would have installed. I did not dig deeper there.

Increasing the timeout for npm install by adding `npm config set fetch-retry-maxtimeout 600000; npm config set fetch-retry-mintimeout 100000` to my script fixed `npm install` but now it ran into a timeout at the gradle build step. Changing the gradle command to `./gradlew assembleRelease -Dorg.gradle.internal.http.socketTimeout=600000 -Dorg.gradle.internal.http.connectionTimeout=600000` fixed that.

This is on a normally decent internet connection. Speed test reports this:

![image](https://github.com/BlueWallet/BlueWallet/assets/96378/54760b73-0a1d-471d-9e51-4405d1aa8c51)

but my provider's performance is at times a bit intermittent. I'm wondering why I consistently get issues with this product but not with others. The number of dependencies? Some config that causes too many to be downloaded in parallel? I don't know ... I wish I understood the difference between podman and docker enough to figure out where to file an issue about the freezing.

**Comment by Overtorment** [comment]
you mentioned docker? where did you get the dockerfile?
